.github/workflows/github_pages.yml

```diff
@@ -0,0 +1,61 @@
+name: Deploy to GitHub Pages
+on:
+  workflow_call:
+    inputs:
+      settings:
+        required: true
+        description: "The path to your Pelican settings file (`pelican`'s `--settings` option), for example: 'publishconf.py'"
+        type: string
+      requirements:
+        required: false
+        default: "pelican"
+        description: "The Python requirements to install, for example to enable markdown and typogrify use: 'pelican[markdown] typogrify'"
+        type: string
+      output-path:
+        required: false
+        default: "output/"
+        description: "Where to output the generated files (`pelican`'s `--output` option)"
+        type: string
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+      - name: Configure GitHub Pages
+        id: pages
+        uses: actions/configure-pages@v3
+      - name: Install requirements
+        run: pip install ${{ inputs.requirements }}
```

Comment:
These feels a bit dodgy: the user could publish a malicious package containing a `pelican` command to PyPI and use `inputs.requirements` to have this workflow install that package, then when the workflow runs `pelican` below it does something malicious. But I think it's okay because the workflow runs in the context of the user's repo not this repo, so the user could only do something malicious to themselves. And this option seems necessary in order to allow the user to install optional things like markdown, typogrify and plugins.